src/capture/pcap_service.c

```diff
@@ -112,16 +112,13 @@ int run_pcap(char *interface, bool immediate, bool promiscuous, int timeout,
   struct bpf_program fp;
   struct pcap_context *ctx = NULL;
 
-  if (!find_device(interface, &net, &mask)) {
-    log_error("find_interfaces fail");
-    return -1;
+  if (find_device(interface, &net, &mask) > 0) {
+    bit32_2_ip((uint32_t)net, ip_str);
+    bit32_2_ip((uint32_t)mask, mask_str);
```

Comment:
Worth doing error checking here?

```suggestion
    if(bit32_2_ip((uint32_t)net, ip_str) == NULL) {
      log_errno("Converting %d to IP failed", net);
      return -1;
    }
    if(bit32_2_ip((uint32_t)mask, mask_str) == NULL) {
      log_errno("Converting %d to IP mask failed", mask);
      return -1;
    }
```